File: templates/node-ts-app/src/money.ts
// Money in integer minor units (cents). Never floats. See standards/domain/finance.md.
export type Cents = number & { readonly __brand: 'Cents' };

export function cents(n: number): Cents {
  if (!Number.isInteger(n)) throw new RangeError('Cents must be an integer (minor units)');
  return n as Cents;
}

/** Sum a list of amounts; total must equal the arithmetic sum (accounting invariant). */
export function total(amounts: readonly Cents[]): Cents {
  return cents(amounts.reduce((a, b) => a + b, 0));
}

export function format(c: Cents, currency = 'USD'): string {
  const sign = c < 0 ? '-' : '';
  const v = Math.abs(c);
  return `${sign}${currency} ${Math.floor(v / 100)}.${String(v % 100).padStart(2, '0')}`;
}

File: templates/python-service/src/app/money.py
"""Money in integer minor units (cents). Never floats. See standards/domain/finance.md."""

from __future__ import annotations

from pydantic import BaseModel, field_validator


class Cents(BaseModel):
    value: int

    @field_validator("value")
    @classmethod
    def _must_be_int(cls, v: int) -> int:
        if not isinstance(v, int) or isinstance(v, bool):
            raise ValueError("Cents must be an integer (minor units)")
        return v


def total(items: list[Cents]) -> Cents:
    """Sum amounts; total equals the arithmetic sum (accounting invariant)."""
    return Cents(value=sum(c.value for c in items))


def fmt(c: Cents, currency: str = "USD") -> str:
    sign = "-" if c.value < 0 else ""
    v = abs(c.value)
    return f"{sign}{currency} {v // 100}.{v % 100:02d}"
